README.md

```diff
@@ -1,2 +1,116 @@
-# notation-ado-task
-Notation azure pipeline task
+# Notation for Azure DevOps Task
+> [!NOTE]
+> The project is under development.
+
+Install Notation CLI, sign or verify container registry artifact.
+
+## Usage
+Notation sign and verify command need the credential to access the registry. User should use [Docker task](https://learn.microsoft.com/azure/devops/pipelines/tasks/reference/docker-v2) to login and the Notation task can get the credential from pipeline context.
+
+For Notation sign command, Azure Key Vault plugin will be used. User should create an Azure Resource Manager serivce connection to grant Azure Key Vault permission for the task.
+
+
+**Prerequisite:**
+1. Create a [Docker Registry service connection](https://learn.microsoft.com/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#docker-registry-service-connection)
+2. Create an [Azure Resource Manager service connection](https://learn.microsoft.com/azure/devops/pipelines/library/service-endpoints?view=azure-devops&tabs=yaml#azure-resource-manager-service-connection) and grant the permission:
+   1. `Open` the created Azure RM service connection and click `Manage Service Principal` to enter the service principal portal.
+   2. The `Application (client) ID` will be used to grant permission for the service principal.
+   3. `Open` Azure Key Vault portal, and enter `Access Policies` page
+   4. `Create` a new policy with `key sign`, `secret get` and `certificate get` permission and grant to the `Application (client) ID` accessed from the previous step.
+
+Then create your own pipeline based on the example.
+
+## Azure Pipeline Example
+**Notation sign**: automatically detect the artifact from previous Docker task
```

Comment:
Can you add an example of `notation sign` without auto detection?

Reply:
Added.